packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -424,6 +424,12 @@ export class StateMachine extends StateMachineBase {
     });
     resource.applyRemovalPolicy(props.removalPolicy, { default: RemovalPolicy.DESTROY });
 
+    // Leaving this commented for review, as I want to make sure this truly isn't necessary.
+    // My understanding is that the `Ref` of the role should be enough to create an implicit dependency.
+    // However, this was adding both a dependency on the role and the policy, but the policy
+    // depends on the state machine thus creating a circular dependency.  By removing
+    // this statement, the circular dependency is no longer an issue.
+    // If removing this is approved in the PR process, I will delete it before merge.
     resource.node.addDependency(this.role);
```

Comment:
I was able to clean this up a little bit and I removed this comment.  The issue here is that Distributed Map states require that the state machine role have execute permissions on the state machine itself.  So this resulted in the policy depending on the state machine, the state machine depending on the state machine role and the state machine role depending on the policy.  Thus, a circular dependency.  I was able to get around this by using some wildcards instead of the specific state machine arn in the policy document.  This is only an issue for the Distributed Map state because I had to add the execution permissions to the policy for the state machine in that case.